pkg/metric/writer_cw.go

```diff
@@ -31,12 +33,15 @@ const (
 	chunkSizeCloudWatch = 20
 	minusOneWeek        = -1 * 7 * 24 * time.Hour
 	plusOneHour         = 1 * time.Hour
+
+	defaultWriteTimeout = 10 * time.Second
 )
 
 type (
 	CloudWatchSettings struct {
-		Naming    CloudwatchNamingSettings `cfg:"naming"`
-		Aggregate bool                     `cfg:"aggregate" default:"true"`
+		Naming       CloudwatchNamingSettings `cfg:"naming"`
+		Aggregate    bool                     `cfg:"aggregate" default:"true"`
+		WriteTimeout time.Duration            `cfg:"write_timeout" default:"10s"`
```

Comment:
It is not a timeout, if a write takes 1 minute but the application is not shutting down, we will still do it, even if the timeout is set to 10s. It is a grace time.